TYPO3.Neos/Resources/Public/JavaScript/Content/Navigate/NodeTree.js

```diff
@@ -276,6 +276,12 @@ define(
 				if (isCurrentNode) {
 					ContentModule.loadPage(node.data.href);
 				}
+
+				var explodedPath = node.data.key.split("@");
+				if (this.get('pageNodePath').indexOf(explodedPath[0]) === 0) {
+					ContentModule.loadPage(node.data.href);
+				}
```

Comment:
a comment here saying it loads the moved page if the currently viewed page is a child of the moved page would be helpful